src/main/java/ui/UI.java

```diff
@@ -221,6 +221,9 @@ private void initPreApplicationState() {
 
         TestController.setUI(this, getParameters());
         prefs = TestController.loadApplicationPreferences();
+        if(TestController.isBypassLogin()){
+            prefs.setLastLoginCredentials("test", "test");
+        }
```

Comment:
This doesn't look like it belongs here. Can we put this inside `loadApplicationPreferences` instead?